### ISSUE TITLE
Using single quotes around abbr value causes error on zsh initialization

### ISSUE THREAD

**Issue by philwhln** [olets/zsh-abbr#10]
I added the following abbreviation with single quotes. It worked fine in that terminal session.

```
❯ abbr -a nodes='echo "MASTERS" && kubectl get nodes --selector="kubernetes.io/role=master" && echo "REGULAR NODES" && kubectl get nodes --selector="kubernetes.io/role!=master,dedicated!=heavy" && echo "HEAVY NODES" && kubectl get nodes --selector="kubernetes.io/role!=master,dedicated=heavy"'
```

When I opened a new terminal I hit this error....

```
[WARNING]: Console output during zsh initialization detected.

abbr add: Expected one argument, got nodes=echo MASTERS && kubectl get nodes --selector=kubernetes.io/role=master && echo REGULAR NODES && kubectl get nodes --selector=kubernetes.io/role!=master,dedicated!=heavy && echo HEAVY NODES && kubectl get nodes --selector=kubernetes.io/role!=master,dedicated=heavy
```

Trying to delete it fails...

```
❯ abbr -e nodes
abbr erase: No matching abbreviation nodes exists
For help run abbr --help
```

**Comment by philwhln** [comment]
Looking in `~/.config/zsh/abbreviations`, I can see the single quotes have been replaced with double-quotes.

```
abbr nodes="echo "MASTERS" && kubectl get nodes --selector="kubernetes.io/role=master" && echo "REGULAR NODES" && kubectl get nodes --selector="kubernetes.io/role!=master,dedicated!=heavy" && echo "HEAVY NODES" && kubectl get nodes --selector="kubernetes.io/role!=master,dedicated=heavy""
```

**Comment by philwhln** [comment]
Editing this file to escape the quotes fixes the error, so the solution would seem to be doing this escaping of double quotes on persisting the values.
```
abbr nodes="echo \"MASTERS\" && kubectl get nodes --selector=\"kubernetes.io/role=master\" && echo \"REGULAR NODES\" && kubectl get nodes --selector=\"kubernetes.io/role!=master,dedicated!=heavy\" && echo \"HEAVY NODES\" && kubectl get nodes --selector=\"kubernetes.io/role!=master,dedicated=heavy\""
```

**Comment by olets** [comment]
Makes sense, thanks for reporting! Yes, the expansions get saved to the user file wrapped in double quotes. The original quote level should be preserved if possible.

It'll be probably a couple weeks before I'm able to look into it. In the meantime, single quotes will work too if you don't want to gunk up your command with slashes, and the echo's will work without quotes. And it's outside my wheelhouse but in at least some circumstances kubectl's --selector doesn't need quotes.

Cool use case, I hadn't thought to use a single abbreviation for a string of commands. I like it.

**Comment by olets** [comment]
Fixed in 3.1.2!

If you version control your user abbreviations file you'll see some noise with this update — more double quotes, more `\`es. You don't have to change the way you write expansions, either interactively or editing the user abbrevations file directly — zsh will figure out the quoting / escaping as needed.
